### PR TITLE
Restore low-priority (0, 1) backends

### PR DESCRIPTION
**Pull request by jaraco**
In #357, we learned that low priority backends are now being excluded from use. That was not an intended change. I think I imagined that only the chainer would be affected, but because the chainer is now in place as the default mechanism for resolving backends, it had the effect of excluding the non-recommended backends.

This change restores support for non-recommended backends.

I think the main reason for adding the limit was to prevent the fail.Keyring from being included, so in this change, backends with a priority of 0 are still excluded from the chained backends.

### CODE REVIEW

**Comment by mitya57** [comment]
printing the backend is still broken:
```
$ python3 -c 'import keyring; print(keyring.get_keyring())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/dmitry/upstream/keyring/keyring/backend.py", line 89, in __str__
    keyring_class.priority))
TypeError: must be real number, not function
```

**Comment by zooba** [comment]
I'd suggest adding `priority = 0` to `ChainerBackend` as well, which should fix the `__str__` and also make sure we don't recursively chain backends.

**Comment by jaraco** [comment]
Good catch!

**Comment by jaraco** [comment]
@zooba I'm thinking about changing the design completely to this:

- extract the chainer backend to its own library.
- users install the chainer backend to opt in to the behavior.
- restore backend discovery to the prior behavior (highest priority is installed).
- chainer has a very high priority (10 maybe).
- chainer backend initializes itself with all viable keyrings (not itself).

This design would have better separation of concerns and give each user/environment more control over how it's used. Keyring might expose an `extra` so you could install `keyring[chained]` or `keyring[multi]`.

How does that approach strike you? How well does it satisfy your use-case?

**Comment by jaraco** [comment]
As soon as I write that, I realize one possible downside - if keyring were to be adopted by stdlib, this behavior is yet another that's not available out of the box. I can't decide if that's an advantage or disadvantage.

**Comment by jaraco** [comment]
Regardless of my comment above, it's probably worth getting a fix out for the issue, so I'll release a 16.0.1 with these changes and those proposed.

**Comment by mitya57** [comment]
I like the proposed idea, it makes the behavior more predictive.

How high is the chance that keyring is adopted by stdlib? In this case some backends won’t be available anyway, because of missing dependencies.

**Comment by zooba** [comment]
I think the chainer is better off in keyring itself, even if it's off by default. But I think as 2FA and password managers gain more appeal, we're more likely to want it on by default.

Only having a single backend forces the tools using it to implement multiple backends - see e.g. pip's handling of netrc and credentials in the URL, both of which could be easily handled by keyring backends.

In terms of putting it in the stdlib, I suspect it would be a requirement that multiple backends be supported. (Obviously I'd vote for that :) ) Alternatively, the setup should be changed so that a higher priority backend is provided with the next priority one, and it can chain manually if it so desires (but without having to parse configuration files itself).

Basically:
* trivial and obvious for a backend to break the chain (return something/raise)
* complex and non-obvious for a sole backend to correctly chain

Means it's better for the core library to handle chaining, and let the backends or configuration break it if they don't want chaining.
